src/analyze/gui/parser.cpp

```diff
@@ -653,7 +658,37 @@ void Parser::parseImpl(const QString& path, const QString& diffBase, const Filte
 
         const auto isReparsing = data == oldData;
 
-        emit progressMessageAvailable(isReparsing ? i18n("reparsing data...") : i18n("parsing data..."));
+        auto parsingMsg = isReparsing ? i18n("reparsing data...") : i18n("parsing data...");
+        emit progressMessageAvailable(parsingMsg);
+
+        const auto numPasses = data->stringCache.diffMode ? 2 : 3;
+        auto updateProgress = [this, numPasses, parsingMsg](std::shared_ptr<const ParserData> const & data, time_point start){
+            auto passCompletion = 1.0*data->parsingState.compressedByte/data->parsingState.fileSize;
```

Comment:
here and below: please put spaces around operators, afaik I even have a .clang-format in the repo that you can use